pkg/elle/core/core.go

```diff
@@ -34,23 +46,41 @@ func CombineExplainer([]DataExplainer) DataExplainer {
 	panic("implement me")
 }
 
+type PathType = Op
+
 type Circle struct {
 	// Eg. [2, 1, 2] means a circle: 2 -> 1 -> 2
-	Path []interface{}
+	Path []PathType
 }
 
 // TODO: refine me
-type Step struct{}
+type Step struct {
+	Result ExplainResult
+}
+
+func explainCyclePairData(pairExplainer DataExplainer, prePath PathType, currentPath PathType) ExplainResult {
```

Comment:
remove useless code?

Reply:
Where is useless code(

Reply:
explainCyclePairData self is useless code.

Reply:
Oh, I'm confused with fp here...